config_copy_123.txt
<?php\
defined('BASEPATH') OR exit('No direct script access allowed');\
\
date_default_timezone_set('Asia/Jakarta');\
//date_default_timezone_set('Asia/Makassar');\
//date_default_timezone_set('Asia/Jayapura');\
\
// $config['base_url'] = '';\
// $config['base_url'] = "http://".$_SERVER['HTTP_HOST'];\
// $config['base_url'] .= preg_replace('@/+$@','',dirname($_SERVER['SCRIPT_NAME'])).'/';\
\
$scheme = (isset($_SERVER["HTTPS"]) && @$_SERVER["HTTPS"] == "on") ? "https" : "http";\
$config['base_url'] = "$scheme://".$_SERVER['HTTP_HOST'];\
$config['base_url'] .= preg_replace('@/+$@','',dirname($_SERVER['SCRIPT_NAME'])).'/';\
\
$config['index_page'] = '';\
\
$config['uri_protocol']\	= 'REQUEST_URI';\
//$config['uri_protocol'] = "ORIG_PATH_INFO";\
$config['url_suffix'] = '';\
$config['language']\	= 'indonesian';\
$config['charset'] = 'UTF-8';\
$config['enable_hooks'] = FALSE;\
//$config['enable_hooks'] = TRUE;\
$config['subclass_prefix'] = 'MY_';\
\
$config['composer_autoload'] = FALSE;\
//$config['permitted_uri_chars'] = 'a-z 0-9~%.:_\\-';\
$config['permitted_uri_chars'] = 'a-z 0-9~%.:_\\-@\\=';\
$config['enable_query_strings'] = FALSE;\
$config['controller_trigger'] = 'c';\
$config['function_trigger'] = 'm';\
$config['directory_trigger'] = 'd';\
$config['allow_get_array'] = TRUE;\
$config['log_threshold'] = 1;\
$config['log_path'] = '';\
$config['log_file_extension'] = '';\
$config['log_file_permissions'] = 0644;\
$config['log_date_format'] = 'Y-m-d H:i:s';\
$config['error_views_path'] = '';\
$config['cache_path'] = '';\
$config['cache_query_string'] = FALSE;\
\
//$config['encryption_key'] = '';\
$config['encryption_key'] = '498f8140edc311eb94b9c83dd4416dc8';\
\
/*\
|--------------------------------------------------------------------------\
| Session Variables\
|--------------------------------------------------------------------------\
|\
| 'sess_driver'\
|\
|\	The storage driver to use: files, database, redis, memcached\
|\
| 'sess_cookie_name'\
|\
|\	The session cookie name, must contain only [0-9a-z_-] characters\
|\
| 'sess_expiration'\
|\
|\	The number of SECONDS you want the session to last.\
|\	Setting to 0 (zero) means expire when the browser is closed.\
|\
| 'sess_save_path'\
|\
|\	The location to save sessions to, driver dependent.\
|\
|\	For the 'files' driver, it's a path to a writable directory.\
|\	WARNING: Only absolute paths are supported!\
|\
|\	For the 'database' driver, it's a table name.\
|\	Please read up the manual for the format with other session drivers.\
|\
|\	IMPORTANT: You are REQUIRED to set a valid save path!\
|\
| 'sess_match_ip'\
|\
|\	Whether to match the user's IP address when reading the session data.\
|\
|\	WARNING: If you're using the database driver, don't forget to update\
|\	         your session table's PRIMARY KEY when changing this setting.\
|\
| 'sess_time_to_update'\
|\
|\	How many seconds between CI regenerating the session ID.\
|\
| 'sess_regenerate_destroy'\
|\
|\	Whether to destroy session data associated with the old session ID\
|\	when auto-regenerating the session ID. When set to FALSE, the data\
|\	will be later deleted by the garbage collector.\
|\
| Other session cookie settings are shared with the rest of the application,\
| except for 'cookie_prefix' and 'cookie_httponly', which are ignored here.\
|\
*/\
$config['sess_driver'] = 'files';\
$config['sess_cookie_name'] = 'ci_session';\
$config['sess_expiration'] = 7200 * 12; // 1 jam\
$config['sess_save_path'] = NULL;\
$config['sess_match_ip'] = FALSE;\
$config['sess_time_to_update'] = 3000;\
$config['sess_regenerate_destroy'] = FALSE;\
\
/*\
|--------------------------------------------------------------------------\
| Cookie Related Variables\
|--------------------------------------------------------------------------\
|\
| 'cookie_prefix'   = Set a cookie name prefix if you need to avoid collisions\
| 'cookie_domain'   = Set to .your-domain.com for site-wide cookies\
| 'cookie_path'     = Typically will be a forward slash\
| 'cookie_secure'   = Cookie will only be set if a secure HTTPS connection exists.\
| 'cookie_httponly' = Cookie will only be accessible via HTTP(S) (no javascript)\
|\
| Note: These settings (with the exception of 'cookie_prefix' and\
|       'cookie_httponly') will also affect sessions.\
|\
*/\
$config['cookie_prefix']\	= '';\
$config['cookie_domain']\	= '';\
$config['cookie_path']\	\	= '/';\
$config['cookie_secure']\	= FALSE;\
$config['cookie_httponly'] \	= FALSE;\
\
/*\
|--------------------------------------------------------------------------\
| Standardize newlines\
|--------------------------------------------------------------------------\
|\
| Determines whether to standardize newline characters in input data,\
| meaning to replace \\r\\n, \\r, \\n occurrences with the PHP_EOL value.\
|\
| WARNING: This feature is DEPRECATED and currently available only\
|          for backwards compatibility purposes!\
|\
*/\
$config['standardize_newlines'] = FALSE;\
\
/*\
|--------------------------------------------------------------------------\
| Global XSS Filtering\
|--------------------------------------------------------------------------\
|\
| Determines whether the XSS filter is always active when GET, POST or\
| COOKIE data is encountered\
|\
| WARNING: This feature is DEPRECATED and currently available only\
|          for backwards compatibility purposes!\
|\
*/\
$config['global_xss_filtering'] = FALSE;\
\
/*\
|--------------------------------------------------------------------------\
| Cross Site Request Forgery\
|--------------------------------------------------------------------------\
| Enables a CSRF cookie token to be set. When set to TRUE, token will be\
| checked on a submitted form. If you are accepting user data, it is strongly\
| recommended CSRF protection be enabled.\
|\
| 'csrf_token_name' = The token name\
| 'csrf_cookie_name' = The cookie name\
| 'csrf_expire' = The number in seconds the token should expire.\
| 'csrf_regenerate' = Regenerate token on every submission\
| 'csrf_exclude_uris' = Array of URIs which ignore CSRF checks\
*/\
//$config['csrf_protection'] = FALSE;\
//$config['csrf_token_name'] = 'csrf_test_name';\
//$config['csrf_cookie_name'] = 'csrf_cookie_name';\
/*\
if (isset($_SERVER["REQUEST_URI"])) {\
    if(stripos($_SERVER["REQUEST_URI"],'/mobile') === FALSE) {\
        $config['csrf_protection'] = TRUE;\
    } else {\
        $config['csrf_protection'] = FALSE;\
    }\
} else {\
    $config['csrf_protection'] = TRUE;\
}\
*/\
$config['csrf_protection'] = TRUE;\
$config['csrf_token_name'] = 'csrf_token';\
$config['csrf_cookie_name'] = 'csrf_cookie';\
$config['csrf_expire'] = 9000;\
$config['csrf_regenerate'] = FALSE;\
$config['csrf_exclude_uris'] = array(\
    'kelasmateri/saveMateri',\
    'kelasmateri/uploadfile',\
    'kelasmateri/deletefile'\
);\
\
/*\
|--------------------------------------------------------------------------\
| Output Compression\
|--------------------------------------------------------------------------\
|\
| Enables Gzip output compression for faster page loads.  When enabled,\
| the output class will test whether your server supports Gzip.\
| Even if it does, however, not all browsers support compression\
| so enable only if you are reasonably sure your visitors can handle it.\
|\
| Only used if zlib.output_compression is turned off in your php.ini.\
| Please do not use it together with httpd-level output compression.\
|\
| VERY IMPORTANT:  If you are getting a blank page when compression is enabled it\
| means you are prematurely outputting something to your browser. It could\
| even be a line of whitespace at the end of one of your scripts.  For\
| compression to work, nothing can be sent before the output buffer is called\
| by the output class.  Do not 'echo' any values with compression enabled.\
|\
*/\
$config['compress_output'] = FALSE;\
\
/*\
|--------------------------------------------------------------------------\
| Master Time Reference\
|--------------------------------------------------------------------------\
|\
| Options are 'local' or any PHP supported timezone. This preference tells\
| the system whether to use your server's local time as the master 'now'\
| reference, or convert it to the configured one timezone. See the 'date\
| helper' page of the user guide for information regarding date handling.\
|\
*/\
$config['time_reference'] = 'local';\
\
/*\
|--------------------------------------------------------------------------\
| Rewrite PHP Short Tags\
|--------------------------------------------------------------------------\
|\
| If your PHP installation does not have short tag support enabled CI\
| can rewrite the tags on-the-fly, enabling you to utilize that syntax\
| in your view files.  Options are TRUE or FALSE (boolean)\
|\
| Note: You need to have eval() enabled for this to work.\
|\
*/\
$config['rewrite_short_tags'] = FALSE;\
\
/*\
|--------------------------------------------------------------------------\
| Reverse Proxy IPs\
|--------------------------------------------------------------------------\
|\
| If your server is behind a reverse proxy, you must whitelist the proxy\
| IP addresses from which CodeIgniter should trust headers such as\
| HTTP_X_FORWARDED_FOR and HTTP_CLIENT_IP in order to properly identify\
| the visitor's IP address.\
|\
| You can use both an array or a comma-separated list of proxy addresses,\
| as well as specifying whole subnets. Here are a few examples:\
|\
| Comma-separated:\	'10.0.1.200,192.168.5.0/24'\
| Array:\	\	array('10.0.1.200', '192.168.5.0/24')\
*/\
$config['proxy_ips'] = '';\
\
$config['min_password_length']  = 6;\
$config['max_password_length'] = 20;\



application/config/config.php
<?php
defined('BASEPATH') OR exit('No direct script access allowed');

date_default_timezone_set('Asia/Jakarta');
//date_default_timezone_set('Asia/Makassar');
//date_default_timezone_set('Asia/Jayapura');

// $config['base_url'] = '';
// $config['base_url'] = "http://".$_SERVER['HTTP_HOST'];
// $config['base_url'] .= preg_replace('@/+$@','',dirname($_SERVER['SCRIPT_NAME'])).'/';

$scheme = (isset($_SERVER["HTTPS"]) && @$_SERVER["HTTPS"] == "on") ? "https" : "http";
$config['base_url'] = "$scheme://".$_SERVER['HTTP_HOST'];
$config['base_url'] .= preg_replace('@/+$@','',dirname($_SERVER['SCRIPT_NAME'])).'/';

$config['index_page'] = '';

$config['uri_protocol']	= 'REQUEST_URI';
//$config['uri_protocol'] = "ORIG_PATH_INFO";
$config['url_suffix'] = '';
$config['language']	= 'indonesian';
$config['charset'] = 'UTF-8';
$config['enable_hooks'] = FALSE;
//$config['enable_hooks'] = TRUE;
$config['subclass_prefix'] = 'MY_';

$config['composer_autoload'] = FALSE;
//$config['permitted_uri_chars'] = 'a-z 0-9~%.:_\-';
$config['permitted_uri_chars'] = 'a-z 0-9~%.:_\-@\=';
$config['enable_query_strings'] = FALSE;
$config['controller_trigger'] = 'c';
$config['function_trigger'] = 'm';
$config['directory_trigger'] = 'd';
$config['allow_get_array'] = TRUE;
$config['log_threshold'] = 1;
$config['log_path'] = '';
$config['log_file_extension'] = '';
$config['log_file_permissions'] = 0644;
$config['log_date_format'] = 'Y-m-d H:i:s';
$config['error_views_path'] = '';
$config['cache_path'] = '';
$config['cache_query_string'] = FALSE;

//$config['encryption_key'] = '';
$config['encryption_key'] = '498f8140edc311eb94b9c83dd4416dc8';

/*
|--------------------------------------------------------------------------
| Session Variables
|--------------------------------------------------------------------------
|
| 'sess_driver'
|
|	The storage driver to use: files, database, redis, memcached
|
| 'sess_cookie_name'
|
|	The session cookie name, must contain only [0-9a-z_-] characters
|
| 'sess_expiration'
|
|	The number of SECONDS you want the session to last.
|	Setting to 0 (zero) means expire when the browser is closed.
|
| 'sess_save_path'
|
|	The location to save sessions to, driver dependent.
|
|	For the 'files' driver, it's a path to a writable directory.
|	WARNING: Only absolute paths are supported!
|
|	For the 'database' driver, it's a table name.
|	Please read up the manual for the format with other session drivers.
|
|	IMPORTANT: You are REQUIRED to set a valid save path!
|
| 'sess_match_ip'
|
|	Whether to match the user's IP address when reading the session data.
|
|	WARNING: If you're using the database driver, don't forget to update
|	         your session table's PRIMARY KEY when changing this setting.
|
| 'sess_time_to_update'
|
|	How many seconds between CI regenerating the session ID.
|
| 'sess_regenerate_destroy'
|
|	Whether to destroy session data associated with the old session ID
|	when auto-regenerating the session ID. When set to FALSE, the data
|	will be later deleted by the garbage collector.
|
| Other session cookie settings are shared with the rest of the application,
| except for 'cookie_prefix' and 'cookie_httponly', which are ignored here.
|
*/
$config['sess_driver'] = 'files';
$config['sess_cookie_name'] = 'ci_session';
$config['sess_expiration'] = 7200 * 12; // 1 jam
$config['sess_save_path'] = NULL;
$config['sess_match_ip'] = FALSE;
$config['sess_time_to_update'] = 3000;
$config['sess_regenerate_destroy'] = FALSE;

/*
|--------------------------------------------------------------------------
| Cookie Related Variables
|--------------------------------------------------------------------------
|
| 'cookie_prefix'   = Set a cookie name prefix if you need to avoid collisions
| 'cookie_domain'   = Set to .your-domain.com for site-wide cookies
| 'cookie_path'     = Typically will be a forward slash
| 'cookie_secure'   = Cookie will only be set if a secure HTTPS connection exists.
| 'cookie_httponly' = Cookie will only be accessible via HTTP(S) (no javascript)
|
| Note: These settings (with the exception of 'cookie_prefix' and
|       'cookie_httponly') will also affect sessions.
|
*/
$config['cookie_prefix']	= '';
$config['cookie_domain']	= '';
$config['cookie_path']		= '/';
$config['cookie_secure']	= FALSE;
$config['cookie_httponly'] 	= FALSE;

/*
|--------------------------------------------------------------------------
| Standardize newlines
|--------------------------------------------------------------------------
|
| Determines whether to standardize newline characters in input data,
| meaning to replace \r\n, \r, \n occurrences with the PHP_EOL value.
|
| WARNING: This feature is DEPRECATED and currently available only
|          for backwards compatibility purposes!
|
*/
$config['standardize_newlines'] = FALSE;

/*
|--------------------------------------------------------------------------
| Global XSS Filtering
|--------------------------------------------------------------------------
|
| Determines whether the XSS filter is always active when GET, POST or
| COOKIE data is encountered
|
| WARNING: This feature is DEPRECATED and currently available only
|          for backwards compatibility purposes!
|
*/
$config['global_xss_filtering'] = FALSE;

/*
|--------------------------------------------------------------------------
| Cross Site Request Forgery
|--------------------------------------------------------------------------
| Enables a CSRF cookie token to be set. When set to TRUE, token will be
| checked on a submitted form. If you are accepting user data, it is strongly
| recommended CSRF protection be enabled.
|
| 'csrf_token_name' = The token name
| 'csrf_cookie_name' = The cookie name
| 'csrf_expire' = The number in seconds the token should expire.
| 'csrf_regenerate' = Regenerate token on every submission
| 'csrf_exclude_uris' = Array of URIs which ignore CSRF checks
*/
//$config['csrf_protection'] = FALSE;
//$config['csrf_token_name'] = 'csrf_test_name';
//$config['csrf_cookie_name'] = 'csrf_cookie_name';
/*
if (isset($_SERVER["REQUEST_URI"])) {
    if(stripos($_SERVER["REQUEST_URI"],'/mobile') === FALSE) {
        $config['csrf_protection'] = TRUE;
    } else {
        $config['csrf_protection'] = FALSE;
    }
} else {
    $config['csrf_protection'] = TRUE;
}
*/
$config['csrf_protection'] = TRUE;
$config['csrf_token_name'] = 'csrf_token';
$config['csrf_cookie_name'] = 'csrf_cookie';
$config['csrf_expire'] = 9000;
$config['csrf_regenerate'] = FALSE;
$config['csrf_exclude_uris'] = array(
    'kelasmateri/saveMateri',
    'kelasmateri/uploadfile',
    'kelasmateri/deletefile',
    'honor/bulk_action'
);

/*
|--------------------------------------------------------------------------
| Output Compression
|--------------------------------------------------------------------------
|
| Enables Gzip output compression for faster page loads.  When enabled,
| the output class will test whether your server supports Gzip.
| Even if it does, however, not all browsers support compression
| so enable only if you are reasonably sure your visitors can handle it.
|
| Only used if zlib.output_compression is turned off in your php.ini.
| Please do not use it together with httpd-level output compression.
|
| VERY IMPORTANT:  If you are getting a blank page when compression is enabled it
| means you are prematurely outputting something to your browser. It could
| even be a line of whitespace at the end of one of your scripts.  For
| compression to work, nothing can be sent before the output buffer is called
| by the output class.  Do not 'echo' any values with compression enabled.
|
*/
$config['compress_output'] = FALSE;

/*
|--------------------------------------------------------------------------
| Master Time Reference
|--------------------------------------------------------------------------
|
| Options are 'local' or any PHP supported timezone. This preference tells
| the system whether to use your server's local time as the master 'now'
| reference, or convert it to the configured one timezone. See the 'date
| helper' page of the user guide for information regarding date handling.
|
*/
$config['time_reference'] = 'local';

/*
|--------------------------------------------------------------------------
| Rewrite PHP Short Tags
|--------------------------------------------------------------------------
|
| If your PHP installation does not have short tag support enabled CI
| can rewrite the tags on-the-fly, enabling you to utilize that syntax
| in your view files.  Options are TRUE or FALSE (boolean)
|
| Note: You need to have eval() enabled for this to work.
|
*/
$config['rewrite_short_tags'] = FALSE;

/*
|--------------------------------------------------------------------------
| Reverse Proxy IPs
|--------------------------------------------------------------------------
|
| If your server is behind a reverse proxy, you must whitelist the proxy
| IP addresses from which CodeIgniter should trust headers such as
| HTTP_X_FORWARDED_FOR and HTTP_CLIENT_IP in order to properly identify
| the visitor's IP address.
|
| You can use both an array or a comma-separated list of proxy addresses,
| as well as specifying whole subnets. Here are a few examples:
|
| Comma-separated:	'10.0.1.200,192.168.5.0/24'
| Array:		array('10.0.1.200', '192.168.5.0/24')
*/
$config['proxy_ips'] = '';

$config['min_password_length']  = 6;
$config['max_password_length'] = 20;
